crates/fluvio-version-manager/src/common/settings.rs

```diff
@@ -0,0 +1,142 @@
+use std::fs::write;
+use std::path::PathBuf;
+
+use anyhow::{Error, Result};
+use serde::{Deserialize, Serialize};
+use semver::Version;
+
+use fluvio_hub_util::fvm::Channel;
+
+use super::workdir::fvm_workdir_path;
+
+pub const SETTINGS_TOML_FILENAME: &str = "settings.toml";
+
+/// The `settings.toml` is in charge of keeping track of the active version
+/// through the default key, which holds the name of the directory under
+/// `~/.fvm/pkgset/default/versions` for the desired default version.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Settings {
+    /// The active `channel` for the Fluvio Installation
+    pub channel: Option<Channel>,
+    /// The specific version in use
+    pub version: Option<Version>,
```

Comment:
While a channel label (stable or latest) is optionally there, isn't there always a version number?

Reply:
Yeah! But as of this point, as no Fluvio version has been installed, both `channel` and `version` will be `None`